fml/logging.cc

```diff
@@ -80,9 +84,35 @@ LogMessage::~LogMessage() {
       priority = ANDROID_LOG_FATAL;
       break;
   }
-  __android_log_write(priority, "flutter", stream_.str().c_str());
-#elif defined(OS_IOS)
-  syslog(LOG_ALERT, "%s", stream_.str().c_str());
+  __android_log_write(priority, "flutter", chars);
+#elif defined(__APPLE__)
+  // Unified logging (os_log) became available in iOS 9.0 and syslog stopped working in iOS 13.0.
+  // idevicesyslog made device syslog available on the connected host, but there is
+  // no known API to view device unified logging on the host.
+  // Flutter tool will continue to observe syslog on devices older than iOS 13.0 since it
+  // provides more logging context, particularly for application crashes.
+  if (__builtin_available(iOS 13.0, macOS 10.11, *)) {
+    static os_log_t engineLog = os_log_create("io.flutter", "engine");
+    switch (severity_) {
+      // LogSeverity LOG_INFO and LOG_WARNING collide with syslog log level macros.
+      case 0 /* LOG_INFO */:
```

Comment:
Please file a bug for us to prefix these macros with `FML_` so we are no longer susceptible to such collisions. Cross referencing that bug with a TODO here would be great. Thanks.